daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/SchemaSetRuntimeData.scala

```diff
@@ -23,20 +23,20 @@ import org.apache.daffodil.processors.unparsers.Unparser
 import org.apache.daffodil.processors.parsers.Parser
 import org.apache.daffodil.processors.TypeCalculatorCompiler.TypeCalcMap
 
-final class SchemaSetRuntimeData(
-  val parser: Parser,
-  val unparser: Unparser,
+final case class SchemaSetRuntimeData(
+  parser: Parser,
+  unparser: Unparser,
   /*
    * Memory of the compiler's warnings. If we save a processor, it's useful to be able
    * to have these warnings.
    */
-  val diagnostics: Seq[Diagnostic],
-  val elementRuntimeData: ElementRuntimeData,
+  diagnostics: Seq[Diagnostic],
+  elementRuntimeData: ElementRuntimeData,
   /*
    * The original variables determined by the schema compiler.
    */
-  variables: VariableMap,
-  val typeCalculators: TypeCalcMap)
+  variables_ : VariableMap,
```

Comment:
I made this a case class to get the automatic copy method. But that also makes all the args into public values. So the _ suffix is to say "not really for public use". 

I think this is moot however, because I'm going to try the approach SL suggested.